docs/pages/architecture/agents.mdx

```diff
@@ -0,0 +1,280 @@
+---
+
+**Teleport agents** route traffic from users through the Teleport Proxy Service
+to and from resources in your infrastructure. This guide describes the
+architecture that enables Teleport to securely manage traffic through this
+pathway.
+
+Teleport agents are running instances of the `teleport` binary, and can run on
+any Linux platform (e.g., bare-metal, a Linux VM, a Docker container, or a pod
+in a Kubernetes cluster). It is up to Teleport administrators to deploy and
+manage agents, including on managed Teleport Enterprise accounts. 
+
+A single agent can run multiple services, and each service can connect to
+multiple target resources. For example, a single `teleport` process running on a
+Linux VM might run both the Teleport SSH Service and Teleport Kubernetes
+Service, which can connect to multiple terminal sessions and Kubernetes
+clusters, respectively.
+
+## Services
+
+An agent can run one or more services. The following agent services are
+available:
+
+| Service | Traffic it proxies |
+|-------|--------------------|
+| SSH Service | SSH traffic to the host where the service runs.|
+| Teleport Kubernetes Service  | HTTPS traffic to and from Kubernetes API servers.|
+| Teleport Database Service    | Database-native wire protocols such as Postgres and MySQL, plus HTTP-based database protocols such as DynamoDB.|
+| Teleport Application Service | HTTPS and TCP forwarding for internal web applications, with optional JSON web tokens. Signed HTTPS messages for cloud provider APIs.|
+| Teleport Desktop Service     | Remote Desktop Protocol traffic to and from RDP servers.|
+
+Agents can also run the Teleport Discovery Service. The Discovery Service
+queries service discovery endpoints to list resources in your infrastructure,
+and configures your Teleport cluster to route traffic to these resources by
+creating dynamic configurations on the Auth Service backend. Agent services
+listed above can then proxy resources enrolled by the Discovery Service.
+
+## Components
+
+![Components of a Teleport agent deployment](../../img/architecture/agent-architecture.png)
+
+A Teleport cluster where an administrator has enrolled resources involves the
+following components:
+
+- **[Teleport Proxy Service](../architecture/proxy.mdx):** A stateless service
+  that performs the function of an authentication gateway, serves the Web UI,
+  and accepts client connections. On managed Teleport Enterprise accounts, this
+  service is accessible at your Teleport account URL, e.g.,
+  `example.teleport.sh`.
+- **[Teleport Auth Service](../architecture/authentication.mdx):** Serves as the
+  cluster's certificate authority, handles user authentication/authorization and
+  issues short-lived client certificates.
+- **Teleport agents:** Agents perform authentication against infrastructure
+  resources, route user traffic to those resources, and perform protocol
+  parsing.
+- **Teleport client tools (`tsh`, Teleport Connect, and the Teleport Web UI):**
```

Comment:
`tbot` ?

Reply:
I haven't mentioned `tbot` at all in this guide, and assume that "client" is the same as "end user". Since this PR has been in the works for a while, I would suggest leaving out `tbot` and adding it in another change if it make sense to!